scripts/irods/database_upgrade.py

```diff
@@ -136,6 +137,11 @@ def run_update(irods_config, cursor):
         if irods_config.catalog_database_type == 'mysql':
             # Rerunning the MySQL script, mysql_functions.sql, fixes issues with sequence numbers
             # and the potential for deadlocks in the database.
+            mysql_exec = shutil.which('mysql')
+            if mysql_exec is None:
+                mysql_exec = shutil.which('mariadb')
+            if mysql_exec is None:
+                raise IrodsError('Database update failed: Could not find \'mysql\' or \'mariadb\' in PATH.')
```

Comment:
This section is run during setup of new servers. Without this change, server setup would fail on systems using a mariadb client without the `mysql` compatibility shim.

Reply:
That makes sense. I guess there's no downside to this since it's enhancing the implementation.

Carry on.

Reply:
Just seeing this.    Yes, this is only initial setup - no upgrade issues as I read it.

Reply:
Cool. Have we ever gone back and modified db schema upgrade logic post release?

Reply:
I don't know that we have.  But this is still the 'latest' schema version, and this code will only fire if the installed schema version was less than 11 and it needed to be brought 'up to be current'... aka an upgrade of iRODS itself on this machine.

This is looking for the mysql or mariadb binary... I think we're okay here.  I can't think of a scenario where this is a breaking change.